extensions/http-curl/client/HTTPClient.cpp

```diff
@@ -330,6 +335,27 @@ void HTTPClient::set_request_method(const std::string method) {
   }
 }
 
+int HTTPClient::onProgress(void *clientp, curl_off_t dltotal, curl_off_t dlnow, curl_off_t ultotal, curl_off_t ulnow){
+  HTTPClient& client = *(HTTPClient*)(clientp);
+  auto now = std::chrono::high_resolution_clock::now();
+  auto elapsed = std::chrono::duration_cast<std::chrono::milliseconds>(now - client.progress_.last_transferred_);
+  if(dlnow != client.progress_.downloaded_data_ || ulnow != client.progress_.uploaded_data_){
+    // did transfer data
+    client.progress_.last_transferred_ = now;
+    client.progress_.downloaded_data_ = dlnow;
+    client.progress_.uploaded_data_ = ulnow;
+    return 0;
+  }
+  // did not transfer data
+  if(elapsed.count() > client.read_timeout_ms_.count()){
```

Comment:
I think linter would ask for some spaces here.

Reply:
will be resolved as part of [MINIFICPP-1227](https://issues.apache.org/jira/browse/MINIFICPP-1227)